Shelter map popup templating and styles

rdev0rigin opened the pull request:
This PR will provide the `_includes/shelter_map.html` with a similar templating and style for the popups as was implemented with `_includes/map.html`.

<img width="814" alt="screen shot 2018-10-08 at 2 34 19 pm" src="https://user-images.githubusercontent.com/26286574/46634869-46553900-cb07-11e8-9aef-68e69c052571.png">

## concern

- Overlaps (two things) 
    - The popup can be over lapped by the filter radio list. 
    - The map width/height takes up the whole screen making it difficult to scroll to the top or bottom of the view. 

<img width="924" alt="screen shot 2018-10-08 at 2 05 55 pm" src="https://user-images.githubusercontent.com/26286574/46634183-f1182800-cb04-11e8-986f-aca3009e81a5.png"> 
  
### suggestion

- side toolbar - If we have a toolbar on the side that has the filter options it could help with scrolling and be out of the way for popups.  

## concern

- More Details - There seems to be more details that we should want to show, like the `special needs` and `accepting`.

### suggestion

- Details View  - Allowing a separate view with the details might be less confusing for the user.

- Slide-Out - A fixed panel at the bottom that can slide up and down might be a good way to provide scrollable content instead of or with a minimal popup.

rdev0rigin commented:
Theres a problem with the controls on mobile views. 
<img width="361" alt="screen shot 2018-10-08 at 5 02 42 pm" src="https://user-images.githubusercontent.com/26286574/46639623-2714d680-cb1c-11e8-893d-ab1a317c0fe6.png">

miklb commented:
Good thing about this is that the storm isn't active and this site isn't being used for live data so we can use it as a staging/test environment at this point. A better UI is definitely a priority before it can be used in a live environment.

rdev0rigin commented:
I refactored the style for mobile for the filter controls.
<img width="397" alt="screen shot 2018-10-08 at 6 56 17 pm" src="https://user-images.githubusercontent.com/26286574/46642535-aa3e2880-cb2c-11e8-8535-b3e86812c045.png">
<img width="403" alt="screen shot 2018-10-08 at 6 56 06 pm" src="https://user-images.githubusercontent.com/26286574/46642536-aa3e2880-cb2c-11e8-8700-bf01c34d6361.png">
<img width="391" alt="screen shot 2018-10-08 at 6 56 29 pm" src="https://user-images.githubusercontent.com/26286574/46642537-aa3e2880-cb2c-11e8-8b36-9d09884e98c1.png">

**Note** I still need to put in the logic for the filters.

omnilord commented on _includes/shelter_map.html:
```diff
@@ -1,125 +1,332 @@
+        window.shelters = sheltersShown = e.target.response.shelters;
+        populateMarkers(sheltersShown);
+    };
+    oReq.open('GET', 'https://irma-api.herokuapp.com/api/v1/shelters', true);
```
Putting a comment here so it's easy to find, the real API end point will be at hurricane-florence-api.herokuapp.com.

miklb commented:
I'm getting an error in the console `ReferenceError: Can't find variable: luxon`

rdev0rigin commented:
> I'm getting an error in the console `ReferenceError: Can't find variable: luxon`

The latest push should fix it, I am getting an error on the Jekyll server though. 
`[2018-10-09 08:11:20] ERROR `/assets/vendor/luxon.js.map' not found.`

Im not sure why it's telling me this. Looking in to it.

rdev0rigin commented:
It looks like a Luxon bug and the builds failing. Im going to swap it out for a MomentJS CDN

rdev0rigin commented:
@miklb Should be fixed now.

miklb commented:
I'm pulling this down now, but quick question? How much of momentJS are we using? Could something like this https://date-fns.org be lighter to load and still get what we need?

rdev0rigin commented:
> I'm pulling this down now, but quick question? How much of momentJS are we using? Could something like this https://date-fns.org be lighter to load and still get what we need?

It looks like we can use that, Ill plunk it in.

miklb commented:
As long as it can serve the same purpose and be loaded modularly and thus lighter. We can always refactor later if it’s a bigger task.
On Oct 9, 2018, 12:58 -0400, Robert Rex Beatie <notifications@github.com>, wrote:
> > I'm pulling this down now, but quick question? How much of momentJS are we using? Could something like this https://date-fns.org be lighter to load and still get what we need?
> It looks like we can use that, Ill plunk it in.
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

rdev0rigin commented:
@miklb Good idea with date-fns. Its in and working.

rdev0rigin commented:
I added the location finder that comes out the (map)box and removed the `NearBy` control.
<img width="324" alt="screen shot 2018-10-09 at 12 17 56 pm" src="https://user-images.githubusercontent.com/26286574/46693017-8b3ba700-cbbd-11e8-9079-24252a05f50d.png">
